modes/ement/evil-collection-ement.el

```diff
@@ -0,0 +1,72 @@
+;;; evil-collection-ement.el --- Evil integration for Ement.el  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2023  Nicholas Vollmer
+
+;; Author:  Nicholas Vollmer
+;; Keywords: convenience
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;;
+
+;;; Code:
+(require 'evil-collection)
+(require 'ement nil t)
+(require 'ement-room-list nil t)
+
+(defgroup evil-collection-ement nil
+  "Evil bindings for Ement.el."
+  :group 'evil-collection)
+
+(defcustom evil-collection-ement-want-auto-retro t
+  "When non-nil, automatically run `ement-room-retro' when `bob' hit."
+  :type 'boolean)
+
+(declare-function evil-previous-line@evil-collection-ement-retro "evil-collection-ement")
+(defun evil-collection-ement-setup ()
+  "Set up `evil' bindings for Ement.el."
+  (evil-collection-define-key 'normal 'ement-describe-room-mode-map
+    (kbd "q") 'quit-window)
+
+  (evil-collection-define-key 'normal 'ement-room-list-mode-map
+    (kbd "RET") 'ement-room-list-RET
+    (kbd "<ret>") 'ement-room-list-RET)
+
+  (evil-collection-define-key '(normal motion) 'ement-room-mode-map
+    (kbd "dd") 'ement-room-delete-message
+    (kbd "D")  'ement-room-delete-message
+    (kbd "a")  'ement-room-transient
+    (kbd "g]") 'ement-room-goto-next
+    (kbd "g[") 'ement-room-goto-prev
+    (kbd "gr") 'ement-room-sync
+    (kbd "gu") 'ement-room-goto-fully-read-marker
+    (kbd "i")  'ement-room-send-message
+    (kbd "I")  'ement-room-compose-message
+    (kbd "q")  'quit-window
+    (kbd "S-RET") 'ement-room-compose-send)
+
+  (when evil-collection-ement-want-auto-retro
```

Comment:
What about `C-u`?

Reply:
Not sure what you mean here.

Reply:
I mean, this covers `k` (line up) but not `C-u` (scroll up).